workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestQuery.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct ComponentInterest.Query objects.
+    /// </summary>
+    public class InterestQuery
+    {
+        private static readonly IReadOnlyList<uint> EmptyList = new List<uint>();
```

Comment:
Remove empty list and instantiate lists on the spot instead.

Reply:
Initially this was done as a List to reduce allocation (because if we don't set an empty list, the field is left as null which then blows up later)

It became ReadOnly because Jamie made the point that it could be modified: https://github.com/spatialos/gdk-for-unity/pull/654#discussion_r251934226 

The tradeoff I see is:

allocate + ensure the lists can't be changed
or
minimise allocation, but run the risk that someone could change the list so it's not actually empty

Reply:
You can't change the read only list but you also aren't using it so it's moot. You are just instantiating a new list each time through slightly obscured syntax right now. If we want to ensure a user can't change the list then we need to make sure the interface given to them is IReadOnlyList.

Reply:
Agreed that we don't want to give users a reference to a mutable internal object.